### PR TITLE
Jasmine's takeaway-challenge

### PR DESCRIPTION
**Pull request by jazzc001**
# Jasmine Cheung

Please write your full name here to make it easier to find your pull request.

# User stories 

Please list which user stories you've implemented (delete the ones that don't apply).

- [x] User story 1: "I would like to see a list of dishes with prices"
- [x] User story 2: "I would like to be able to select some number of several available dishes"
- [x] User story 3: "I would like to check that the total I have been given matches the sum of the various dishes in my order"
- [ ] User story 4: "I would like to receive a text such as "Thank you! Your order was placed and will be delivered before 18:52" after I have ordered"

# README checklist

Does your README contains instructions for

- [ ] how to install,
- [ ] how to run,
- [ ] and how to test your code?

[Here is a pill](https://github.com/makersacademy/course/blob/main/pills/readmes.md) that can help you write a great README!

### CODE REVIEW

File: lib/menu.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class Menu
+  DISHES = {
+    "bruschetta" => 3,
+    "spaghetti carbonara" => 8,
+  }
+
```
Comment (r94o) [inline lib/menu.rb:6]: Good to use a constant here as the dishes will not change

File: lib/menu.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  def dish?(dish)
+    DISHES.key?(dish)
+  end
+
+  def price(dish)
+    raise "'#{dish}' is not on the menu" unless dish?(dish)
+
```
Comment (r94o) [inline lib/menu.rb:13]: nice guard clause, very readable

File: lib/order.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+    @dishes[dish] += quantity
+  end
+
+  def total
+    return 0 if @dishes.empty?
+
+    subtotals = @dishes.map { |dish, quantity| @menu.price(dish) * quantity }
+    subtotals.sum
+  end
```
Comment (r94o) [inline lib/order.rb:21]: Think could can replace the .map block with .sum block and therefore be able to remove line 20

File: lib/order.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+  def place
+    raise "cannot place an empty order" if @dishes.empty?
+
```
Comment (r94o) [inline lib/order.rb:25]: Good edge case checking

**Review by r94o — COMMENTED** [review]
Nice one! Tests are isolated and cover edge cases. Code is fairly readable. I assume message_sender is another (to be implemented) class which will hold the Twilio code?

The only thing I would suggest is creating a dish class so DISHES could store an array of the Dish class so you could then just reference the price by Dish.price rather than DISHES["spaghetti carbonara"] which may make it slightly more readable
